src/NHibernate/Multi/LinqBatchItem.cs

```diff
@@ -90,27 +99,24 @@ private List<T> GetTransformedResults(IList transformerList)
 			};
 		}
 
-		private System.Type GetResultTypeIfChanged()
+		private IList GetTypedResults(System.Type type)
 		{
-			if (_postExecuteTransformer == null)
-			{
-				return null;
-			}
-			var elementType = _postExecuteTransformer.Method.GetParameters()[1].ParameterType.GetGenericArguments()[0];
-			if (typeof(T).IsAssignableFrom(elementType))
```

Comment:
There is no need to do such analysis - if type is overridden its type provided in `ExpressionToHqlTranslationResults.ExecuteResultTypeOverride`